packages/esm-reports-app/src/components/overview.component.tsx

```diff
@@ -0,0 +1,350 @@
+import ReportStatus from './report-status.component';
+import { COMPLETED, FAILED, PROCESSING, RAN_REPORT_STATUSES, REQUESTED, SAVED } from './report-statuses-constants';
+import ReportOverviewButton from './report-overview-button.component';
+import { PRIVILEGE_SYSTEM_DEVELOPER } from '../constants';
+
+const OverviewComponent: React.FC = () => {
+  const { t } = useTranslation();
+  const currentSession = useSession();
+
+  let [checkedReportUuidsArray, setCheckedReportUuidsArray] = useState([]);
+  const [downloadReportButtonVisible, setDownloadReportButtonVisible] = useState(false);
+
+  useEffect(() => {
+    setDownloadReportButtonVisible(checkedReportUuidsArray.length > 0);
+  }, [checkedReportUuidsArray]);
+
+  const tableHeaders = [
+    { key: 'reportName', header: t('reportName', 'Report Name') },
+    { key: 'status', header: t('status', 'Status') },
+    { key: 'requestedBy', header: t('requestedBy', 'Requested by') },
+    { key: 'requestedOn', header: t('requestedOn', 'Requested on') },
+    { key: 'outputFormat', header: t('outputFormat', 'Output format') },
+    { key: 'parameters', header: t('parameters', 'Parameters') },
+    { key: 'actions', header: t('actions', 'Actions') },
+  ];
+
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE_NUMBER);
+  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+
+  const { reports, reportsTotalCount, mutateReports } = useReports(
+    RAN_REPORT_STATUSES.join(','),
+    currentPage,
+    pageSize,
+  );
+
+  const layout = useLayoutType();
+
+  function getReportStatus(row) {
+    return row?.cells.find((cell) => cell.info?.header === 'status')?.value;
+  }
+
+  function isCurrentUserTheSameAsReportRequestedByUser(reportRequestUuid: string) {
+    const report = reports.find((tableRow) => tableRow.id === reportRequestUuid);
+    const requestedByUserUuid = report?.requestedByUserUuid;
+    const currentUserUuid = currentSession?.user.uuid;
+
+    return requestedByUserUuid === currentUserUuid;
+  }
+
+  function isSystemDeveloperUser() {
+    return userHasAccess(PRIVILEGE_SYSTEM_DEVELOPER, currentSession.user);
+  }
+
+  function isEligibleReportUser(reportRequestUuid: string) {
+    return isCurrentUserTheSameAsReportRequestedByUser(reportRequestUuid) || isSystemDeveloperUser();
+  }
+
+  function renderRowCheckbox(row, index) {
+    const statusCell = row?.cells.find((cell) => cell.info?.header === 'status');
+    const statusValue = statusCell?.value;
+    if (statusValue === COMPLETED || statusValue === SAVED) {
+      return (
+        <td className={index % 2 == 0 ? styles.rowCellEven : styles.rowCellOdd}>
+          <Checkbox
+            id={`checkbox-${row.id}`}
+            onChange={(e) => handleOnCheckboxClick(e)}
+            checked={checkedReportUuidsArray.includes(row.id)}
+          />
+        </td>
+      );
+    } else {
+      return <td className={index % 2 == 0 ? styles.rowCellEven : styles.rowCellOdd}></td>;
```

Comment:
Prefer using classnames for conditionally applying styles.

Reply:
fixed